File: Middleware/StockMarket/StockMarket.AdminAPI/Services/AdminService.cs
using StockMarket.AdminAPI.Models;
using StockMarket.AdminAPI.Repository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace StockMarket.AdminAPI.Services
{
    public class AdminService:IAdminService
    {
        private IAdminRepository repo;

        public AdminService(IAdminRepository _repo) {
            repo = _repo;
        }
        public void AddCompany(Company company)
        {
            repo.AddCompany(company);
        }

        public void EditCompany(Company company)
        {
            repo.EditCompany(company);
        }

        public void DeleteCompany(string companyCode)
        {
            repo.DeleteCompany(companyCode);
        }

        public void UpdateIpo(IpoDetails ipo)
        {
            repo.UpdateIpo(ipo);
        }

        public void AddIpo(IpoDetails item)
        {
            repo.AddIpo(item);
        }

        public void DeleteIpo(int Id)
        {
            repo.DeleteIpo(Id);
        }

        public List<Company> GetAllCompany()
        {
            return repo.GetAllCompany();
        }

        public List<IpoDetails> GetAllIpo()
        {
            return repo.GetAllIpo();
        }
    }
}

File: UI/StockMarketInterface/src/app/Shared/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from "rxjs";
import { Company } from '../Models/company';
import { IpoDetails } from '../Models/ipo-details';
@Injectable({
  providedIn: 'root'
})
export class AdminService {
  constructor(private http:HttpClient) { }
  path: string = "http://localhost:51789/Admin/";

  public GetAllCompany(): Observable<Company[]> {
    return this.http.get<Company[]>(this.path + "GetCompany")
  }

  public GetAllIpo(): Observable<IpoDetails[]> {
    return this.http.get<IpoDetails[]>(this.path + "GetIpo")
  }

  public AddCompany(item: Company): Observable<any> {
    console.log(item);
    return this.http.post<any>(this.path + 'AddCompany',item);
  }
  public UpdateCompany(item:Company): Observable<any> {
    console.log(item);
    return this.http.put<any>(this.path + 'EditCompany' , item);
  }
  public DeleteCompany(code: string): Observable<any> {
    console.log(code);
    return this.http.delete<any>(this.path + 'DeleteCompany/' + code);
  }
  public AddIpo(item: IpoDetails): Observable<any> {
    console.log(item);
    return this.http.post<any>(this.path + 'AddIpo', item);
  }
  public UpdateIpo(item: IpoDetails): Observable<any> {
    console.log(item);
    return this.http.put<any>(this.path + 'UpdateIpo', item);
  }
  public DeleteIpo(id: number): Observable<any> {
    console.log(id);
    return this.http.delete<any>(this.path + 'DeleteIpo/' + id);
  }
}
